server/src/main/java/io/crate/expression/scalar/HasSchemaPrivilegeFunction.java

```diff
@@ -270,20 +271,29 @@ private Collection<Privilege.Type> normalizePrivilegeIfLiteral(Symbol symbol) {
     public final Boolean evaluate(TransactionContext txnCtx, NodeContext nodeCtx, Input[] args) {
         Object userNameOrOid, schemaNameOrOid, privileges;
 
+        var sessionUser = USER_BY_NAME.apply(nodeCtx.userLookup(), txnCtx.sessionSettings().userName());
+        User user;
         if (args.length == 2) {
-            userNameOrOid = txnCtx.sessionSettings().userName();
             schemaNameOrOid = args[0].value();
             privileges = args[1].value();
+            user = sessionUser;
         } else {
             userNameOrOid = args[0].value();
+            if (userNameOrOid == null) {
+                return null;
+            }
+            user = getUser.apply(nodeCtx.userLookup(), userNameOrOid);
+            // Only superusers can call this function for other users
+            if (user.name().equals(sessionUser.name()) == false && sessionUser.isSuperUser() == false) {
+                throw new MissingPrivilegeException(user.name());
```

Comment:
this looks like we are saying that "other user doesn't have permission" - shouldn't we use session user here? So that we convey a message "current user cannot check privileges of other users"

Reply:
yes, sorry, I saw it but forgot to change, thx!